File: ng-shop-api/Models/Image.cs
namespace ng_shop_api.Models
{
    public class Image
    {
        public int Id { get; set; }
        public string Url { get; set; }
        public bool IsThumbnail { get; set; }
        public string PublicId { get; set; }

        public int? ProductId { get; set; }
        public Product Product { get; set; }

        public int? BrandId {get; set;}
        public Brand Brand { get; set; }
    }
}

File: ng-shop-spa/src/app/_models/image.ts
import { Product } from './product';
import { Brand } from './brand';

export interface Image {
    id: number;
    url: string;
    isThumbnail: boolean;
    isBanner: boolean;
    publicId: string;
    productId: number;
    brandID: number;

    product: Product;
    brand: Brand;
}
